AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Subscribe/DataStoreObserveQueryOperation.swift

```diff
@@ -142,6 +142,10 @@ class ObserveQueryTaskRunner<M: Model>: InternalTaskRunner, InternalTaskAsyncThr
             if let batchItemsChangedSink = batchItemsChangedSink {
                 batchItemsChangedSink.cancel()
             }
+            
+            if let readyEventSink = readyEventSink {
+                        readyEventSink.cancel()
+            }
```

Comment:
in `resetState()` below, set the sink to nil, so that we stop observing the modelSyncedEvent after it has been reset

Reply:
ya makes sense, done !